tests/unit/hive_metastore/test_views_sequencer.py

```diff
@@ -1,65 +1,85 @@
 import json
+from itertools import chain
 from pathlib import Path
+from typing import TypeVar
+from unittest.mock import create_autospec
 
 import pytest
 from databricks.labs.lsql.backends import MockBackend, SqlBackend
 
 from databricks.labs.ucx.hive_metastore import TablesCrawler
-from databricks.labs.ucx.hive_metastore.views_migrator import ViewsMigrator
+from databricks.labs.ucx.hive_metastore.mapping import Rule, TableToMigrate
+from databricks.labs.ucx.hive_metastore.views_sequencer import TableMigrationSequencer
 
 SCHEMA_NAME = "schema"
 
+T = TypeVar('T')
+
+
+def flatten(lists: list[list[T]]) -> list[T]:
+    return list(chain.from_iterable(lists))
+
 
 def test_migrate_no_view_returns_empty_sequence():
     samples = Samples.load("db1.t1", "db2.t1")
     sql_backend = mock_backend(samples, "db1", "db2")
     crawler = TablesCrawler(sql_backend, SCHEMA_NAME, ["db1", "db2"])
-    migrator = ViewsMigrator(crawler)
-    sequence = migrator.sequence()
+    tables = [TableToMigrate(table, create_autospec(Rule)) for table in crawler.snapshot()]
```

Comment:
as I understand it a data class in Python has a different meaning than a Kotlin one or a Java record. It's more a Lombok kind of thing i.e. syntactic sugar to avoid boilerplate code. `Rule` is a full fledged class (I'll give you it only has 3 very lean methods).
In Java I would also mock it. I don't find that weird, but I could create a dummy global `Rule` used by all tests.

Reply:
oh, i think of Python data classes as Scala case classes - those can have methods